components/updates/filters/UpdateSearchFilter.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { omitBy } from 'lodash';
+import { useRouter } from 'next/router';
+import { FormattedMessage } from 'react-intl';
+import styled from 'styled-components';
+
+import Container from '../../Container';
+import { Box } from '../../Grid';
+import SearchBar from '../../SearchBar';
+
+const FilterLabel = styled.label`
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 16px;
+  letter-spacing: 0.06em;
+  text-transform: uppercase;
+  color: #4e5052;
+  margin-bottom: 10px;
+`;
+
+const SearchFormContainer = styled(Box)`
+  width: 100%;
+  min-width: 10rem;
+`;
+
+const ROUTE_PARAMS = ['collectiveSlug', 'offset'];
+
+const UpdateSearchFilter = () => {
+  const router = useRouter() || {};
+  const { collectiveSlug, searchTerm } = router.query;
+  const updateFilters = queryParams => {
+    return router.push({
```

Comment:
Having routing here is a problem: what if we want to re-use this component on another page? What if we want to test it in isolation?

On other pages, filters have a callback and let the parent handle the routing. See https://github.com/opencollective/opencollective-frontend/blob/57f80ac3ebea94bd4e9282e5c0aca63ab6131a98/components/transactions/filters/TransactionsTypeFilter.js for example

Reply:
@Betree : I agree. I've refactored all the routing logic and simplified the code. Let me know how it looks now. 😄